File: go/subscription_event_data.go
// this file is @generated
package meteroid

import "time"

type SubscriptionEventData struct {
	// RFC 3339 timestamp.
	ActivatedAt *time.Time `json:"activated_at,omitempty"`

	AutoAdvanceInvoices bool `json:"auto_advance_invoices"`

	BillingDayAnchor int32 `json:"billing_day_anchor"`

	BillingStartDate *string `json:"billing_start_date,omitempty"`

	// Present on `subscription.cancelled` when a reason was supplied.
	CancellationReason *string `json:"cancellation_reason,omitempty"`

	ChangeType *SubscriptionUpdateType `json:"change_type,omitempty"`

	ChargeAutomatically bool `json:"charge_automatically"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Currency string `json:"currency"`

	// User-defined custom property values, keyed by definition key.
	CustomProperties RequiredMap[any] `json:"custom_properties"`

	CustomerAlias *string `json:"customer_alias,omitempty"`

	CustomerId CustomerId `json:"customer_id"`

	CustomerName string `json:"customer_name"`

	EndDate *string `json:"end_date,omitempty"`

	InvoiceMemo *string `json:"invoice_memo,omitempty"`

	InvoiceThreshold *string `json:"invoice_threshold,omitempty"`

	MrrCents int64 `json:"mrr_cents"`

	NetTerms int32 `json:"net_terms"`

	Period BillingPeriodEnum `json:"period"`

	PlanName string `json:"plan_name"`

	PurchaseOrder *string `json:"purchase_order,omitempty"`

	StartDate string `json:"start_date"`

	Status SubscriptionStatusEnum `json:"status"`

	SubscriptionId SubscriptionId `json:"subscription_id"`

	TrialDuration *int32 `json:"trial_duration,omitempty"`

	Version int32 `json:"version"`
}

File: python/meteroid/models/subscription_event_data.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .billing_period_enum import BillingPeriodEnum
from .customer_id import CustomerId
from .subscription_id import SubscriptionId
from .subscription_status_enum import SubscriptionStatusEnum
from .subscription_update_type import SubscriptionUpdateType


@dataclasses.dataclass
class SubscriptionEventData(BaseModel):
    auto_advance_invoices: bool

    billing_day_anchor: int

    charge_automatically: bool

    created_at: datetime

    currency: str

    custom_properties: t.Any
    """User-defined custom property values, keyed by definition key."""

    customer_id: CustomerId

    customer_name: str

    mrr_cents: int

    net_terms: int

    period: BillingPeriodEnum

    plan_name: str

    start_date: str

    status: SubscriptionStatusEnum

    subscription_id: SubscriptionId

    version: int

    activated_at: t.Optional[datetime] = None

    billing_start_date: t.Optional[str] = None

    cancellation_reason: t.Optional[str] = None
    """Present on `subscription.cancelled` when a reason was supplied."""

    change_type: t.Optional[SubscriptionUpdateType] = None

    customer_alias: t.Optional[str] = None

    end_date: t.Optional[str] = None

    invoice_memo: t.Optional[str] = None

    invoice_threshold: t.Optional[str] = None

    purchase_order: t.Optional[str] = None

    trial_duration: t.Optional[int] = None
